activemq-broker/src/main/java/org/apache/activemq/broker/region/DestinationStatistics.java

```diff
@@ -252,7 +242,29 @@ public void setParent(DestinationStatistics parent) {
             maxUncommittedExceededCount.setParent(null);
             networkEnqueues.setParent(null);
             networkDequeues.setParent(null);
+            // [AMQ-9437] Advanced Message Statistics do not parent.
         }
     }
 
+    // This is the only method that can mutate the messageFlowStats state
+    // so we only need to synchronize this method to make sure we don't have 2 threads
+    // trying to set at the same time.
+    public synchronized void setAdvancedMessageStatisticsEnabled(boolean enabled) {
+        // we can just use the get() here on the reference as no other spot can
```

Comment:
You can get rid of this comment here after this was refactored

Reply:
Comment removed via commit c079d1bd03df5a2faf35f2ec9625e7e51b2a4973